src/Build/Utilities/StreamLineSpanReader.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Build.Utilities;
+
+/// <summary>
+/// Reads lines of text from a <see cref="Stream"/> into <see cref="ReadOnlySpan{T}"/>s for further processing.
+/// Allows efficient, low-allocation consumption of textual data from a stream.
+/// </summary>
+internal sealed class StreamLineSpanReader
```

Comment:
@stephentoub Do we have anything like this in .NET Core? Basically ReadOnlySpan<char> producing StreamReader?

Reply:
Nope.  We have a method that enumerates the lines in a span, but no helper for enumerating the lines from a Stream as spans.  We could consider adding something like that to StreamReader, e.g. a variant of ReadLine that returned a span. It's in the gray area of the kinds of things we're comfortable doing on mainline types, as the span that's handed out would have its contents overwritten by subsequent reads, but it's the kind of thing we'll do when there are good motivating scenarios.
cc: @GrabYourPitchforks

Reply:
@drewnoakes just wrote one for solution parsing, right?